src/main/java/me/ryanhamshire/GPFlags/util/ChatUtil.java

```diff
@@ -0,0 +1,225 @@
+                message = message.replace(clickComponent, "");
+                requiredChanges = true;
+            }
+            matcher = pattern.matcher(message);
+
+            if (!requiredChanges) {
+                break;
+            }
+        }
+
+        // Check for hover-components
+        pattern = Pattern.compile("<hover:.*?>");
+        matcher = pattern.matcher(message);
+        while (matcher.find()) {
+            boolean requiredChanges = false;
+
+            String hoverComponent = message.substring(matcher.start(), matcher.end());
+            if (hoverComponent.contains("show_text") && !sender.hasPermission("gpflags.messages.hover.show_text")) {
+                message = message.replace(hoverComponent, "");
+                requiredChanges = true;
+            }
+            if (hoverComponent.contains("show_item") && !sender.hasPermission("gpflags.messages.hover.show_item")) {
+                message = message.replace(hoverComponent, "");
+                requiredChanges = true;
+            }
+            if (hoverComponent.contains("show_entity") && !sender.hasPermission("gpflags.messages.hover.show_entity")) {
+                message = message.replace(hoverComponent, "");
+                requiredChanges = true;
+            }
+            matcher = pattern.matcher(message);
+
+            if (!requiredChanges) {
+                break;
+            }
+        }
+
+        // Check for hex-colors
+        message = message.replace("<#", "<color:#");
+        pattern = Pattern.compile("<color:#.*?>");
+        matcher = pattern.matcher(message);
+        while (matcher.find()) {
+            boolean requiredChanges = false;
+            String hexComponent = message.substring(matcher.start(), matcher.end());
+            if (!sender.hasPermission("gpflags.messages.hex-colors")) {
+                message = message.replace(hexComponent, "");
+                requiredChanges = true;
+            }
+            matcher = pattern.matcher(message);
+
+            if (!requiredChanges) {
+                break;
+            }
+        }
+
+        // Check for colors
+        pattern = Pattern.compile("<color:.*?>");
+        matcher = pattern.matcher(message);
+        while (matcher.find()) {
+            boolean requiredChanges = false;
+            String colorComponent = message.substring(matcher.start(), matcher.end());
+            if (!sender.hasPermission("gpflags.messages.colors")) {
+                message = message.replace(colorComponent, "");
+                requiredChanges = true;
+            }
+            matcher = pattern.matcher(message);
+
+            if (!requiredChanges) {
+                break;
+            }
+        }
+
+        // Check for bold, italic, strikethrough, underline, obfuscated
+        pattern = Pattern.compile("<(bold|italic|strikethrough|underlined|obfuscated)>");
```

Comment:
minimessage tags are case insensitive so this wont catch stuff that are like <BOLD>